src/main/java/com/exasol/UserInput.java

```diff
@@ -0,0 +1,194 @@
+        private Goal goal;
+        private Set<Platform.PlatformName> platforms;
+        private String repositoryName;
+        private String repositoryOwner;
+
+        /**
+         * Add a git branch.
+         *
+         * @param gitBranch name of a branch to work with.
+         * @return builder instance for fluent programming
+         */
+        // [impl->dsn~users-can-set-git-branch-for-validation~1]
+        public Builder gitBranch(final String gitBranch) {
+            this.gitBranch = gitBranch;
+            return this;
+        }
+
+        /**
+         * Add a run goal.
+         *
+         * @param goal run goal. Supported goals: release, validate
+         * @return builder instance for fluent programming
+         */
+        // [impl->dsn~users-set-run-goal~1]
+        public Builder goal(final String goal) {
+            this.goal = Goal.getGoal(goal);
+            return this;
+        }
+
+        /**
+         * Add release platforms.
+         *
+         * @param platforms one or more platforms for validation or release. Supported values: github
+         * @return builder instance for fluent programming
+         */
+        // [impl->dsn~users-set-release-platforms~1]
+        public Builder platforms(final String... platforms) {
+            this.platforms = Platform.PlatformName.toSet(platforms);
+            return this;
+        }
+
+        /**
+         * Add a name of a repository.
+         *
+         * @param repositoryName name of a target project from GitHub
+         * @return builder instance for fluent programming
+         */
+        // [impl->dsn~users-set-project~1]
+        public Builder repositoryName(final String repositoryName) {
+            this.repositoryName = repositoryName;
+            return this;
+        }
+
+        /**
+         * Add an owner of a GitHub repository.
+         *
+         * @param repositoryOwner an owner of a GitHub repository
+         * @return builder instance for fluent programming
+         */
+        public Builder repositoryOwner(final String repositoryOwner) {
+            this.repositoryOwner = repositoryOwner;
+            return this;
+        }
+
+        /**
+         * Create a new {@link UserInput} instance.
+         *
+         * @return new {@link UserInput} instance
+         */
+        public UserInput build() {
+            validateMandatoryParameters();
+            validateGoalAndBranch();
+            return new UserInput(this);
+        }
+
+        private void validateGoalAndBranch() {
+            if (this.goal == Goal.RELEASE && this.gitBranch != null) {
+                throw new IllegalArgumentException(
+                        "E-RR-1: Please, remove branch parameter if you want to make a release.");
+            }
+        }
+
+        private void validateMandatoryParameters() {
+            if (this.goal == null) {
+                throwExceptionForMissingParameter("E-RR-2", "goal");
+            }
```

Comment:
```suggestion
            Objects.requireNonNull(this.goal , () -> getExceptionForMissingParameter("E-RR-2", "goal"));
```

and change `throwExceptionForMissingParameter` --> `getExceptionForMissingParameter`

Reply:
It looks very elegant but doesn't work. Maybe I did something wrong, we can discuss it